src/components/Home/WelcomeHeader.js

```diff
@@ -37,6 +48,94 @@ const WelcomeHeader = () => {
       },
     },
   };
+  
+  useEffect(() => {
+    if (localPath !== ''){
+      const installSampleProtocol = async() => {
+        return openNetcanvas(localPath)
+      }
+      installSampleProtocol()
+    }
+  }, [localPath])
+
+  const getURL = uri =>
+  new Promise((resolve, reject) => {
+    try {
+      resolve(new URL(uri));
+    } catch (error) {
+      reject(error);
+    }
+  });
+
+const urlError = friendlyErrorMessage("The location you gave us doesn't seem to be a valid URL. Check the location, and try again.");
+const networkError = friendlyErrorMessage("We weren't able to fetch your protocol. Your device may not have an active network connection, or you may have mistyped the URL. Ensure you are connected to a network, double check your URL, and try again.");
+const fileError = friendlyErrorMessage('The protocol could not be saved to your device. You might not have enough storage available. ');
+
+/**
+ * Download a protocol from a remote server.
+ *
+ * If the URL points to an instance of a Network Canvas Server, then the caller must ensure
+ * that the SSL certificate has been trusted. See {@link ApiClient#addTrustedCert}.
+ *
+ * @param {string} uri
+ * @return {string} output filepath
+ */
+const downloadProtocolFromURI = inEnvironment((environment)=> {
+
+  if (environment === environments.ELECTRON) {
+    const request = require('request-promise-native');
+    const path = require('path');
+    const electron = require('electron');
+    const { dialog } = electron.remote
+    const tempPath = (electron.app || electron.remote.app).getPath('temp');
+    const from = path.join(tempPath, 'SampleProtocol') + '.netcanvas';
```

Comment:
You should not hard code the name of the protocol here. This will cause a collision/overwrite and unpredictable results with the code is run multiple times.

You could either use a uuid, or you could use the protocol name. If you do the latter, you must handle the case where the temp directory path (including the filename) already exists.